server/app/views/applicant/ProgramCardViewRenderer.java

```diff
@@ -225,7 +225,9 @@ public LiTag createProgramCard(
     // Use external link if it is present else use the default Program details page
     String programDetailsLink =
         program.externalLink().isEmpty()
-            ? applicantRoutes.show(profile, applicantId, program.id()).url()
+            ? profile.isPresent() && applicantId.isPresent()
+                ? applicantRoutes.show(profile.get(), applicantId.get(), program.id()).url()
+                : applicantRoutes.show(program.id()).url()
```

Comment:
It's great that those `show` and `review` routes already existed without applicant id!